packages/vscode-extension/src/chat/chat.ts

```diff
@@ -0,0 +1,160 @@
+Answer any and all user questions regarding React Native.\n\n
+Always assume that the user already has a development environment set up for React Native and Expo.\n\n
+`;
+const START_OF_USER_QUESTION = "\n# ANSWER THE FOLLOWING USER QUESTION:\n\n";
+
+const BASE_RADON_AI_URL = "https://radon-ai-backend.swmansion.com";
+
+interface IChatResult extends vscode.ChatResult {
+  metadata: {
+    command: string;
+  };
+}
+
+export function registerChat(context: vscode.ExtensionContext) {
+  const handler: vscode.ChatRequestHandler = async (
+    request: vscode.ChatRequest,
+    chatContext: vscode.ChatContext,
+    stream: vscode.ChatResponseStream,
+    token: vscode.CancellationToken
+  ): Promise<IChatResult> => {
+    stream.progress("Thinking...");
+    Logger.info("Chat requested");
+
+    const jwt = await getLicenseToken();
+
+    if (!jwt) {
+      Logger.warn("No license found. Please activate your license.");
+      getTelemetryReporter().sendTelemetryEvent("chat:requested:no-license");
+
+      stream.markdown(
+        "You need to have a valid license to use the Radon AI Chat. Please activate your license."
+      );
+      return { metadata: { command: "" } };
+    }
+
+    try {
+      let json;
+      try {
+        const url = new URL("/api/documentation", BASE_RADON_AI_URL);
+        const response = await fetch(url, {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+            "Authorization": `Bearer ${jwt}`,
+          },
+          body: JSON.stringify({ prompt: request.prompt }),
+        });
+
+        if (response.status !== 200) {
+          Logger.error(`Failed to fetch response from Radon AI with status: ${response.status}`);
+          getTelemetryReporter().sendTelemetryEvent("chat:error", {
+            error: `Failed to fetch with status: ${response.status}`,
+          });
+        }
+
+        json = await response.json();
+      } catch (error) {
+        if (error instanceof Error) {
+          Logger.error(error.message);
+          getTelemetryReporter().sendTelemetryEvent("chat:error", { error: error.message });
+        } else {
+          Logger.error(String(error));
+        }
+      }
+
+      let systemPrompt = "";
+
+      if (json) {
+        systemPrompt += START_OF_DOCUMENTATION + json.docs + END_OF_DOCUMENTATION;
+      }
+
+      const previousResponses = chatContext.history.filter(
+        (chatTurn) => chatTurn.participant === CHAT_PARTICIPANT_ID
+      );
+
+      if (previousResponses.length > 0) {
+        let history = "";
+        previousResponses.forEach((response) => {
+          if ("prompt" in response) {
+            history += `USER: ${response.prompt}\n\n`;
+          }
+          if ("response" in response) {
+            response.response.forEach((r) => {
+              if (r instanceof vscode.ChatResponseMarkdownPart) {
+                history += `ASSISTANT: ${r.value.value}\n\n`;
+              }
+            });
+          }
```

Comment:
NIT: If `response` is meant for assistant's messages, maybe we should rename
- `previousResponses` => `chatMessageHistory`
- `response` => `chatMessage`?

Reply:
Fixed in c42e402c